docs/core/addresses.md

```diff
@@ -0,0 +1,62 @@
+---
+id: addresses
+title: Generating addresses
+---
+
+Almost all Bitcoin applications need to generate addresses
+for their users somehow. There's a lot going on in getting
+a correct bitcoin address, but our APIs make it possible to
+to get started with all types of addresses in a matter of
+minutes.
+
+## Generating SegWit (bech32) addresses
+
+Generating native SegWit addresses in the bech32 format
+is something that all Bitcoin applications should enable,
+as it makes the transaction fees less expensive, and also
+makes the addresses more readable by humans. However, it
+has seen slower than necessary adoption. With Bitcoin-S
+you can generate bech32 addresses in four(!) lines of code
+(not counting comments and imports), so now there's no
+reason to keep using legacy transaction formats.
+
+```scala mdoc:to-string
+import org.bitcoins.core.{crypto, protocol, config}
+// if you want to get addresses for mainnet, just import
+// config.MainNet here instead
+import config.TestNet3
+import crypto.ECPrivateKey
+
+// this gets all addresses into scope
+import protocol._
+
+// this gets all scriptPubKeys into scope
+import protocol.script._
+
+// this generates a random private key
+val privkey = ECPrivateKey()
+val pubkey = privkey.publicKey
+
+val segwitAddress = {
+    // see https://bitcoin.org/en/glossary/pubkey-script
+    // for reading resources on the details of scriptPubKeys
+    // pay-to-witness-pubkey-hash scriptPubKey V0
+    val scriptPubKey = P2WPKHWitnessSPKV0(pubkey)
+    Bech32Address(scriptPubKey, TestNet3)
+}
+```
+
+## Generating legacy (base58) addresses
```

Comment:
Just because it's so much simpler I'd argue this section should go above segwit

Reply:
We want to push people towards segwit though! And That's also one of the differentiating things with Bitcoin-S, how easy segwit addresses are. Notice that the code in the block below reuses the values from the block above, so the only lines that are different are 
```scala
val scriptPubKey = P2WPKHWitnessSPKV0(pubkey)
    Bech32Address(scriptPubKey, TestNet3)
```
and 
```scala
 val legacyAddress = P2PKHAddress(pubkey, TestNet3)
```

Reply:
Ah, good points :)